app/src/main/resources/log4j2.xml

```diff
@@ -0,0 +1,30 @@
+<?xml version='1.0' encoding='UTF-8' ?>
+
+<configuration>
+
+    <Properties>
+<!--        <Property name="logFile">"${sys:user.home}/.corda/cli/logs/corda-cli.log"</Property>-->
+    </Properties>
+
+    <Appenders>
+        <!-- Logs the redirected stdOut and stdErr to the console -->
+        <Console name="LogToConsole" target="SYSTEM_OUT"/>
+
+        <File name="LogToFile" fileName="${sys:logFile:-${sys:user.home}/.corda/cli/logs/corda-cli.log}" append="true">
+            <PatternLayout pattern="%d{yyyy-MM-dd HH:mm:ss.SSS} %level [%t] [%c] - %msg%n"/>
+        </File>
+
+    </Appenders>
+
+    <Loggers>
+        <!-- avoid duplicated logs with additivity=false -->
+        <Logger name="org.pf4j" level="info" additivity="false">
+            <AppenderRef ref="LogToFile"/>
+        </Logger>
+        <Root level="info">
+            <AppenderRef ref="LogToConsole"/>
```

Comment:
If I do:
```
LoggerFactory.getLogger(MyClass.class).info("My message")
```
I would expect this log line to be only sent to a file but not to CLI console.
What would prevent this log line for appearing on CLI console, please?

Reply:
that will end up on the console too. Maybe the way that system out and err are captured needs to be looked at so we can avoid this.

Reply:
I think we can remove `LogToConsole` from `Root` and add two more loggers like so:
```
<Logger name="System Out" level="info" additivity="false">
            <AppenderRef ref="LogToConsole"/>
            <AppenderRef ref="LogToFile"/>
</Logger>

<Logger name="System Err" level="info" additivity="false">
            <AppenderRef ref="LogToConsole"/>
            <AppenderRef ref="LogToFile"/>
</Logger>
```
This should ensure that only asomething logged using `System Out` and `System Err` gets written to CLI console. But we will always log everything into a file.

Reply:
for some reason log4j wouldnt respect LogToFile if the level was set to the systems, I have updated the code now, but do we want to always log every system.out.println to the file too?

Reply:
> do we want to always log every system.out.println to the file too?

Yes, that would make sense to me. If anything is going wrong with CLI command - we can ask for **1** log file which will have everything we need: stdOut, stdErr and log lines from all the other logger along with timestamps. So we would be able to establish exactly what went on and in which sequence.

Reply:
done